app/controllers/concerns/prominence_headers.rb

```diff
@@ -0,0 +1,34 @@
+##
+# Set any headers related to prominence.
+#
+module ProminenceHeaders
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_prominence_headers
+  end
+
+  def has_prominence
+    raise NotImplementedError
+  end
+
+  private
+
+  def set_prominence_headers
+    return unless has_prominence
+    send("#{ has_prominence.prominence }_headers")
```

Comment:
Maybe a minor rename since the called method is actually setting the headers, rather than returning a hash for something else to use?

```diff
- send("#{ has_prominence.prominence }_headers")
+ send("set_#{ has_prominence.prominence }_headers")
```